### ISSUE TITLE
 Exception in callback of async function: { Error: socket hang up

### ISSUE THREAD

**Issue by hermes-pimentel** [RocketChat/Rocket.Chat#15116]
Hi guys, after migrate to version 1.2.3 all my servers after a hour or so running start logging this message and then stop. 

RocketChat: Exception in callback of async function: { Error: connect EADDRNOTAVAIL 127.0.0.1:3000 - Local (127.0.0.1:0)
RocketChat: at Object._errnoException (util.js:992:11)
RocketChat: at _exceptionWithHostPort (util.js:1014:20)
RocketChat: at internalConnect (net.js:960:16)
RocketChat: at defaultTriggerAsyncIdScope (internal/async_hooks.js:284:19)
RocketChat: at GetAddrInfoReqWrap.emitLookup [as callback] (net.js:1106:9)
RocketChat: at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:97:10)
RocketChat: code: 'EADDRNOTAVAIL',
RocketChat: errno: 'EADDRNOTAVAIL',
RocketChat: syscall: 'connect',
RocketChat: address: '127.0.0.1',
RocketChat: port: 3000 }

RocketChat: Exception in callback of async function: { Error: socket hang up
RocketChat: at createHangUpError (_http_client.js:331:15)
RocketChat: at Socket.socketCloseListener (_http_client.js:363:23)
RocketChat: at emitOne (events.js:121:20)
RocketChat: at Socket.emit (events.js:211:7)
RocketChat: at TCP._handle.close [as _onclose] (net.js:557:12) code: 'ECONNRESET' }
RocketChat: Exception in callback of async function: { Error: connect EADDRNOTAVAIL 127.0.0.1:3000 - Local (127.0.0.1:0)
RocketChat: at Object._errnoException (util.js:992:11)
RocketChat: at _exceptionWithHostPort (util.js:1014:20)
RocketChat: at internalConnect (net.js:960:16)
RocketChat: at defaultTriggerAsyncIdScope (internal/async_hooks.js:284:19)
RocketChat: at GetAddrInfoReqWrap.emitLookup [as callback] (net.js:1106:9)
RocketChat: at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:97:10)
RocketChat: code: 'EADDRNOTAVAIL',
RocketChat: errno: 'EADDRNOTAVAIL',
RocketChat: syscall: 'connect',
RocketChat: address: '127.0.0.1',
RocketChat: port: 3000 }
RocketChat: Exception in callback of async function: { Error: socket hang up
RocketChat: at createHangUpError (_http_client.js:331:15)
RocketChat: at Socket.socketCloseListener (_http_client.js:363:23)
RocketChat: at emitOne (events.js:121:20)
RocketChat: at Socket.emit (events.js:211:7)
RocketChat: at TCP._handle.close [as _onclose] (net.js:557:12) code: 'ECONNRESET' }

Using lsof I see a lot of TCP connections like this:

node      2739 3209   rocketchat *993u     IPv4             185890       0t0        TCP localhost:41785->localhost:hbci (CLOSE_WAIT)
node      2739 3209   rocketchat *995u     IPv4             185893       0t0        TCP localhost:41787->localhost:hbci (CLOSE_WAIT)

Someone have a hint?

**Comment by reetp** [comment]
Can you please help the developers by filling out a proper bug report? They can't really help without this information
Use this link: https://github.com/RocketChat/Rocket.Chat/issues/new?template=bug_report.md

Anything else special? Proxies, bots, webhooks?

**Comment by hermes-pimentel** [comment]
Hi, after I put the logs to debug I found the issue, but still doest understand what have changed or what I should do:

```
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Wrong instance, proxing to: localhost:3000
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Upload URL: /ufs/AmazonS3:Uploads/4BpeTqXHFLwHNgqXN?token=2b0be9397B&progress=0.5454424395765364
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Wrong instance, proxing to: localhost:3000
RocketChat: server.js:207 #033[34mMeteor ➔ method#033[39m getRoomRoles -> userId: KsgAgHPr3BX7K4oQR , arguments: [ 'KsgAgHPr3BX7K4oQRrgaCmgh2WozkkksGJ' ]
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Upload URL: /ufs/AmazonS3:Uploads/4BpeTqXHFLwHNgqXN?token=2b0be9397B&progress=0.5454424395765364
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Wrong instance, proxing to: localhost:3000
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Upload URL: /ufs/AmazonS3:Uploads/4BpeTqXHFLwHNgqXN?token=2b0be9397B&progress=0.5454424395765364
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Wrong instance, proxing to: localhost:3000
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Upload URL: /ufs/AmazonS3:Uploads/4BpeTqXHFLwHNgqXN?token=2b0be9397B&progress=0.5454424395765364
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Wrong instance, proxing to: localhost:3000
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Upload URL: /ufs/AmazonS3:Uploads/4BpeTqXHFLwHNgqXN?token=2b0be9397B&progress=0.5454424395765364
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Wrong instance, proxing to: localhost:3000
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Upload URL: /ufs/AmazonS3:Uploads/4BpeTqXHFLwHNgqXN?token=2b0be9397B&progress=0.5454424395765364
RocketChat: server.js:207 #033[34mUploadProxy ➔ debug#033[39m Wrong instance, proxing to: localhost:3000
```

Looks like that during an upload to S3 the instance start to connect to localhost until a socket error due all file descriptor get used. 

My S3 configuration:

<img width="1404" alt="Screen Shot 2019-08-07 at 10 25 04" src="https://user-images.githubusercontent.com/7027009/62626431-c69e2380-b8fd-11e9-9045-40fe88bae9a5.png">
<img width="1291" alt="Screen Shot 2019-08-07 at 10 25 14" src="https://user-images.githubusercontent.com/7027009/62626434-c867e700-b8fd-11e9-8a20-24c8fd85603e.png">

**Comment by hermes-pimentel** [comment]
My Current Setup is:

10 Ec2 instances, with a Auto Scaling Group,  running behind a ALB with session affinity. 
Mongo Version: 4.0.9
Linux CentOS

What Broadcast Connected Instances mean? This configuration is ok?

<img width="851" alt="Screen Shot 2019-08-07 at 11 02 05" src="https://user-images.githubusercontent.com/7027009/62629222-277c2a80-b903-11e9-9b25-565261b96eee.png">

**Comment by hermes-pimentel** [comment]
I found a workaround:

I had to add this two lines in my service (systemctl):

```
PermissionsStartOnly=true
ExecStartPre=/bin/bash -c "/bin/systemctl set-environment INSTANCE_IP=$(/bin/hostname -I)"
```

With this, all my instances are connected in broadcast connected instance ( Info page, in administrator console). And all uploads work as expected.

I still doesnt understand why rocket enter in a loop, when try to proxy a request and fall back to 127.0.0.1. Maybe a is bug, and probably should be investigated.
